boilerplates/react-union-boilerplate-liferay-ssr/liferay/ssr-filter/src/main/java/eu/reactunion/ssr/filter/ServerSideRenderingFilter.java

```diff
@@ -0,0 +1,120 @@
+package eu.reactunion.ssr.filter;
+
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.servlet.BaseFilter;
+import com.liferay.portal.kernel.servlet.BufferCacheServletResponse;
+import feign.Feign;
+import org.osgi.service.component.annotations.Component;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.PrintWriter;
+
+@Component(
+        immediate = true,
+        property = {
+                "dispatcher=REQUEST",
+                "dispatcher=FORWARD",
+                "servlet-context-name=",
+                "servlet-filter-name=Server-side Rendering Filter",
+                "url-pattern=/*",
+                "after-filter=Cache Filter"
+        },
+        service = Filter.class
+)
+public class ServerSideRenderingFilter extends BaseFilter {
+    private static final Log _log = LogFactoryUtil.getLog(ServerSideRenderingFilter.class);
+
+    private static final String SSR_FLAG = "SSR";
+
+    private final RenderingServerApi renderingServerApi = Feign
+            .builder()
+            .target(RenderingServerApi.class, "http://localhost:3303");
+
+    private boolean renderingServerUp = true;
```

Comment:
be careful, it must be thread-safe.  you cannot set/assign this parameter from filter methods. Need to have synchronized set and get methods, or it must be AtomicBoolean and its set and get methods must be used.